src/chainstate/stacks/db/blocks.rs

```diff
@@ -163,11 +163,68 @@ pub enum MemPoolRejection {
     BadAddressVersionByte,
     NoCoinbaseViaMempool,
     NoSuchChainTip(BurnchainHeaderHash,BlockHeaderHash),
-    WrongNetwork,
     DBError(db_error),
     Other(String),
 }
 
+impl MemPoolRejection {
+    pub fn into_json(self, txid: &Txid) -> serde_json::Value {
+        use self::MemPoolRejection::*;
+        let (reason_code, reason_data) = match self {
+            SerializationFailure(e) => ("Serialization", 
+                                        Some(json!({"message": e.to_string()}))),
+            DeserializationFailure(e) => ("Deserialization",
+                                          Some(json!({"message": e.to_string()}))),
+            FailedToValidate(e) => ("SignatureValidation",
+                                    Some(json!({"message": e.to_string()}))),
+            FeeTooLow(actual, expected) => ("FeeTooLow", 
+                                            Some(json!({
+                                                "expected": expected,
+                                                "actual": actual}))),
+            BadNonces(TransactionNonceMismatch {
+                expected, actual, principal, is_origin, .. }) =>
+                ("BadNonce",
+                 Some(json!({
+                     "expected": expected,
+                     "actual": actual,
+                     "principal": principal.to_string(),
+                     "is_origin": is_origin}))),
+            NotEnoughFunds(expected, actual) => 
+                ("NotEnoughFunds",
+                 Some(json!({
+                     "expected": format!("0x{}", to_hex(&expected.to_be_bytes())),
+                     "actual": format!("0x{}", to_hex(&actual.to_be_bytes()))
```

Comment:
In `NotEnoughFunds(expected, actual)`, it looks like `expected` and `actual` are quantities of tokens, why are we returning a `0x...` hex string?

Reply:
Because the fund amount here is a u128 -- which would break many (if not all) JSON parsers if it were a JSON number (even though the JSON spec doesn't place a max limit on numbers). Here and other places where u128s are returned (like `/v2/accounts/<>`) return a hex value so that the consumer knows they need to do _something_ special to get a number.